sky/packages/sky/lib/src/gestures/lsq_solver.dart

```diff
@@ -0,0 +1,190 @@
+import "dart:math" as math;
+import "dart:typed_data";
+
+class Vector {
+  Vector(int size)
+  : _offset = 0, _length = size, _elem = new Float64List(size);
+
+  Vector.fromValues(List<double> values)
+  : _offset = 0, _length = values.length, _elem = values;
+
+  Vector.fromVOL(List<double> values, int offset, int length)
+  : _offset = offset, _length = length, _elem = values;
+
+  int get length => _length;
+
+  operator [](int i) => _elem[i + _offset];
+  operator []=(int i, double value) => _elem[i + _offset] = value;
+
+  operator *(Vector a) {
+    double result = 0.0;
+    for (int i = 0; i < _length; i++) {
+      result += this[i] * a[i];
+    }
+    return result;
+  }
+
+  double norm() => math.sqrt(this * this);
+
+  String toString() {
+    String result = "";
+    for (int i = 0; i < _length; i++) {
+      if (i > 0)
+        result += ", ";
+        result += this[i].toString();
+    }
+    return result;
+  }
+
+  final int _offset;
+  final int _length;
+  final List<double> _elem;
+}
+
+class Matrix {
```

Comment:
Should Matrix and Vector be private to this file?  We probably don't want to cause confusion with our use of Matrix4 elsewhere for transform matricies.